File: LMS/LMS.UI/huda-center/src/app/shared/filters/hr/leave.filter.ts
// src/app/shared/filters/hr/leave-filter.ts

import { LeaveStatus } from '../../models/hr/leaves/leave-status.enum';
import { LeaveType } from '../../models/hr/leaves/leave-type.enum';
import { Filter } from '../base/filter'; 
export enum LeaveOrderBy {
  ByEmployeeName = 0,
  ByStartDate = 1,
  ByEndDate = 2,
  ByStatus = 3,
  ByType = 4,
  ByCreatedAt = 5
}

export interface LeaveFilter extends Filter {
  status?: LeaveStatus | null; // int? في الباك إند
  type?: LeaveType | null; // int? في الباك إند
  isPaid?: boolean | null;
  isDesc?: boolean;
  orderBy?: LeaveOrderBy;
}

File: LMS/LMS.Application/Filters/HR/EmployeeIdentity/LeaveFilter.cs
using LMS.Application.Filters.Base;
using LMS.Domain.HR.Enums;

namespace LMS.Application.Filters.HR.EmployeeIdentity
{
    public class LeaveFilter : Filter
    {
        public Guid? EmployeeId { get; set; } = null;
        public int? Status { get; set; }
        public bool? IsPaid { get; set; }
        public bool IsDesc { get; set; } = true;
        public LeaveType? Type { get; set; }
        public LeaveOrderBy OrderBy { get; set; } = LeaveOrderBy.ByCreatedAt;
    }
}